runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftOverlay.java

```diff
@@ -78,35 +80,56 @@ public Dimension render(Graphics2D graphics)
 		for (WidgetItem item : widgetItems)
 		{
 			Varbits varbits;
+			int maxEssence;
 
 			switch (item.getId())
 			{
 				case ItemID.SMALL_POUCH:
 					varbits = Varbits.POUCH_SMALL;
+					maxEssence = 3;
 					break;
 				case ItemID.MEDIUM_POUCH:
 				case MEDIUM_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_MEDIUM;
+					maxEssence = 6;
 					break;
 				case ItemID.LARGE_POUCH:
 				case LARGE_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_LARGE;
+					maxEssence = 9;
 					break;
 				case ItemID.GIANT_POUCH:
 				case GIANT_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_GIANT;
+					maxEssence = 12;
 					break;
 				default:
 					continue;
 			}
 
 			final Rectangle bounds = item.getCanvasBounds();
 			final TextComponent textComponent = new TextComponent();
+			int essenceInPouch = client.getSetting(varbits);
+
+			if (essenceInPouch == 0)
+			{
+				graphics.setColor(new Color(255, 0, 0, 50));
```

Comment:
I'd personally make the Colors constants at the top;
```
private static final Color RED = new Color(255, 0, 0, 50);
private static final Color GREEN = new Color(0, 255, 0, 50);
private static final Color YELLOW = new Color(255, 255, 0, 50);
```
And reference them here.

I think also a switch statement would look cleaner here;
```
switch (essenceInPouch)
{
    case 0:
        graphics.setColor(RED);
        break;
    case maxEssence:
        graphics.setColor(GREEN);
        break;
    case default:
        graphics.setColor(YELLOW);
        break;
}
```

Reply:
And not name them RED, GREEN and YELLOW but actually name them to what they represent

Reply:
Dont you need constant case not dynamic for switch?

Reply:
Why not use the Color.RED, Color.GREEN, Color.YELLOW constants provided by Java?

Reply:
Transparency